owncloudApp/src/main/java/com/owncloud/android/presentation/ui/authentication/LoginActivity.kt

```diff
@@ -370,6 +375,35 @@ class LoginActivity : AppCompatActivity(), SslUntrustedCertDialog.OnSslUntrusted
         })
     }
 
+    /**
+     * Register client if possible.
+     */
+    private fun registerClient(
+        authorizationEndpoint: Uri,
+        registrationEndpoint: String
+    ) {
+        oauthViewModel.registerClient(registrationEndpoint)
+        oauthViewModel.registerClient.observe(this, {
+            when (it.peekContent()) {
+                is UIResult.Loading -> {
+                }
+                is UIResult.Success -> {
+                    Timber.d("Client registered: ${it.peekContent().getStoredData()}")
+                    it.peekContent().getStoredData()?.let { clientRegistrationInfo ->
+                        performGetAuthorizationCodeRequest(
+                            authorizationEndpoint,
+                            clientRegistrationInfo.clientId
+                        )
+                    }
+                }
+                is UIResult.Error -> {
+                    Timber.e(it.peekContent().getThrowableOrNull(), "Client registration failed.")
+                    Toast.makeText(this, it.peekContent().getThrowableOrNull().toString(), Toast.LENGTH_SHORT).show()
```

Comment:
What about creating an activity extension for this and just call `showError`?

Reply:
Ups, if that operation fails, we don't need to print the error, but follow the login flow.